src/mon/NVMeofGwMon.cc

```diff
@@ -0,0 +1,530 @@
+                std::stringstream  sstrm1;
+                sstrm1 << state.availability;
+                f->dump_string("Availability", sstrm1.str());
+                sstrm1.str("");
+                for (size_t i = 0; i < map.created_gws[group_key].size(); i++) {
+                    sstrm1 << " " << anas[i]+1 <<": " << state.sm_state[anas[i]] << ",";
+                }
+                f->dump_string("ana states", sstrm1.str());
+                f->close_section();
+            }
+            f->flush(rdata);
+            sstrm.str("");
+        }
+        else {
+            sstrm << "num_gws  0";
+        }
+        getline(sstrm, rs);
+        mon.reply_command(op, err, rs, rdata, get_last_committed());
+        return true;
+    }
+    return false;
+}
+
+bool NVMeofGwMon::prepare_command(MonOpRequestRef op)
+{
+    dout(10)  << dendl;
+    auto m = op->get_req<MMonCommand>();
+    int rc;
+    std::stringstream sstrm;
+    bufferlist rdata;
+    string rs;
+    int err = 0;
+    cmdmap_t cmdmap;
+    bool response = false;
+
+    if (!cmdmap_from_json(m->cmd, &cmdmap, sstrm))
+    {
+        string rs = sstrm.str();
+        mon.reply_command(op, -EINVAL, rs, rdata, get_last_committed());
+        return true;
+    }
+
+    string format = cmd_getval_or<string>(cmdmap, "format", "plain");
+    boost::scoped_ptr<Formatter> f(Formatter::create(format));
+
+    const auto prefix = cmd_getval_or<string>(cmdmap, "prefix", string{});
+
+    dout(10) << "MonCommand : "<< prefix <<  dendl;
+    if( prefix == "nvme-gw create" || prefix == "nvme-gw delete" ) {
+        std::string id, pool, group;
+
+        cmd_getval(cmdmap, "id", id);
+        cmd_getval(cmdmap, "pool", pool);
+        cmd_getval(cmdmap, "group", group);
+        auto group_key = std::make_pair(pool, group);
+        dout(10) << " id "<< id <<" pool "<< pool << " group "<< group << dendl;
+        if(prefix == "nvme-gw create"){
+            rc = pending_map.cfg_add_gw(id, group_key);
+            if(rc == -EINVAL){
+                err = rc;
+                dout (4) << "Error: GW cannot be created " << id << " " << pool << " " << group << "  rc " << rc << dendl;
+                sstrm.str("");
+            }
+        }
+        else{
+            rc = pending_map.cfg_delete_gw(id, group_key);
+            if(rc == -EINVAL){
+                dout (4) << "Error: GW not found in the database " << id << " " << pool << " " << group << "  rc " << rc << dendl;
+                err = 0;
+                sstrm.str("");
+            }
+        }
+        if((rc != -EEXIST) && (rc != -EINVAL)) //propose pending would be generated by the PaxosService
+            response = true;
+    }
+
+    getline(sstrm, rs);
+    if(response == false) {
+       if (err < 0 && rs.length() == 0)
+       {
+         rs = cpp_strerror(err);
+         dout(1) << "Error command  err : "<< err  << " rs-len: " << rs.length() <<  dendl;
```

Comment:
Too-high debug. A badly-formed client command does not require this level of error logging. dout(10) would be acceptable.